services/pane.test.js

```diff
@@ -156,54 +157,107 @@ describe(dirname, function () {
       });
     });
 
-    describe('openValidationErrors', function () {
-      var fn = lib[this.title],
-        sandbox;
+    describe('openNewPage', function () {
+      var mock = {
+          locals: {edit: true}
+        },
+        noFilter = function () {},
+        createPage, el, env, getState, result, sandbox, spy, stub, template;
 
       before(function () {
-        var template = document.createElement('template'),
-          headEl = document.createElement('div'),
-          paneInnerEl = document.createElement('div'),
-          toolbar = document.createElement('div');
-
-        // add template
-        template.classList.add('kiln-pane-template');
-        headEl.classList.add('pane-header');
-        paneInnerEl.classList.add('pane-inner');
-        template.content.appendChild(headEl);
-        template.content.appendChild(paneInnerEl);
-        document.body.appendChild(template);
-
-        // add toolbar
-        toolbar.classList.add('kiln-toolbar');
-        document.body.appendChild(toolbar);
+        env = new nunjucks.Environment();
+        // satisfy request for nunjucks filter in template
+        env.addFilter('includeFile', noFilter); // TODO – include nunjucks filters
+        template = env.getPreprocessedTemplate('template.nunjucks');
       });
 
       beforeEach(function () {
+        result = template.render(mock);
         sandbox = sinon.sandbox.create();
         sandbox.stub(ds);
-        sandbox.stub(state, 'get');
+        getState = sandbox.stub(state, 'get');
       });
 
       afterEach(function () {
+        document.body.innerHTML = undefined;
+        el = undefined;
         sandbox.restore();
       });
 
-      it('opens an error pane', function () {
-        lib.close();
+      it('has a template skeleton for selecting page type', function () {
+        document.body.innerHTML += result;
+        el = document.querySelector('.new-page-actions');
+        expect(el).to.exist;
+      });
+
+      it('has a toolbar button for opening the page type selection dialog', function () {
+        document.body.innerHTML += result;
+        el = document.querySelector('.kiln-toolbar-button.new');
+        expect(el).to.exist;
+      });
 
-        function expectPane() {
-          expect(document.querySelector('.pane-inner .error-message')).to.not.equal(null);
+      it('creates a clone of the pane template and adds it to the dom on [+ page] button click', function () {
+        getState.returns(Promise.resolve({}));
+        lib.close();
+        document.body.innerHTML += result;
+        el = document.querySelector('.kiln-toolbar-button.new');
+        spy = sandbox.spy(el, 'click');
+        el.click();
+
+        function clickToOpenPane() {
+          expect(el).to.exist;
+          sinon.assert.called(spy);
```

Comment:
It's dependent on el existing, but that's nonsense! Thanks for the catch – updating.